cmd/protoc-gen-go-grpc/grpc.go

```diff
@@ -87,7 +87,11 @@ func generateFile(gen *protogen.Plugin, file *protogen.File) *protogen.Generated
 		return nil
 	}
 	filename := file.GeneratedFilenamePrefix + "_grpc.pb.go"
-	g := gen.NewGeneratedFile(filename, file.GoImportPath)
+	path := file.GoImportPath
+	if *standaloneDir {
+		path = "."
```

Comment:
And if the purpose is to have different go packages (directories) for messages and RPC stubs. Would defining messages and RPCs in different proto files (and different packages) be better? That way the generated code is consistent with the proto files, and proto files are the single source of truth.

Reply:
1) i don't think that passing package name is better than set ".", not sure about file descriptor and server reflection
2) creating different proto files for messages and rpc stubs not so useful for small micro-services and also in this case - you grpc file descriptor will be different and so like in 1) - not big difference.

so i don't see any drawbacks to have such option. but maybe somebody who knows about server reflection can check this?

Reply:
@dsnet do you have recommendation on the value of `goImportPath` for `NewGeneratedFile`?
Would `"."` be an OK value?

Reply:
cc @neild since I believe @dsnet has left the project.

Reply:
The rules for how input `.proto` files map to Go import paths and generated output files are quite complex. I would be extremely wary of making any changes here without fully understanding how it fits into the current behaviors.

It sounds like your goal is to place the service definitions (the `_grpc.pb.go` file) into a separate Go package from the other types (the `.pb.go`).

An import path of `"."` is wrong. You want the actual import path. A dummy value like `"."` might work in some cases, but will produce surprising results in others. You also probably need to make the `protogen` package aware of the import path used, to get the filename right. (The example here uses `paths=source_relative`, but how does this work when `paths=import` or `module=${MODULE}`?) 

A question is how the user specifies the import path. Import paths are usually specified in one of two ways: A `go_package` option in the `.proto` file, or a `M` flag on the command line. If you want a separate import path for the services, I think you should add equivalent mechanisms to specify them. Perhaps a `go_grpc_package` option in the `.proto` file, or a `go_services_package` repeated option allows specifying the name of the service; and perhaps a gRPC-specific equivalent to `M`.

I'm happy to review a design doc if someone wants to write one.

I would not recommend touching this behavior in an ad hoc PR without design review. As I said, it's really extremely complicated.

Reply:
Thanks for reply, @neild!

I was suggesting making the command line flag in this PR a package name string, instead of a boolean. But I just realized that it only works for one proto file (or proto files in the same package).

A `go_grpc_package` option would be nice. The downside is that the proto files need to be modified (or maybe that's an advantage as now the proto file has all the information).
I think if we move along this path, we might need both the `go_grpc_package` option for proto files, __and__ a flag similar to `M`, just like what's done in the message codegen.